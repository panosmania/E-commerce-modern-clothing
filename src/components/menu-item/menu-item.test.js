import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

import { MenuItem } from "./menu-item.component";

describe("MenuItem component", () => {
  let wrapper;
  let mockHistory;
  let mockMatch;
  const linkUrl = "/hats";
  const size = "large";
  const imageUrl = "testimage";

  beforeEach(() => {
    mockMatch = {
      path: "/shop",
    };

    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      title: "hats",
      imageUrl: imageUrl,
      size: size,
      linkUrl: linkUrl,
      history: mockHistory,
      match: mockMatch,
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it("should render MenuItem component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call history.push with the right string when div with classname menu-item clicked", () => {
    wrapper.find(".menu-item").simulate("click");
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  it("should pass size to MenuItemContainer as the prop size", () => {
    expect(wrapper.find(".background-image").prop("style")).toHaveProperty(
      "backgroundImage",
      `url(${imageUrl})`
    );
  });
});
