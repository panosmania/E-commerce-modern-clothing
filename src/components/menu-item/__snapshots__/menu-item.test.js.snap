// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render MenuItem component 1`] = `
<div
  className="(large ? '380px' : '240px') menu-item"
  onClick={[Function]}
>
  <div
    className="background-image"
    style={
      Object {
        "backgroundImage": "url(testimage)",
      }
    }
  />
  <div
    className="content"
  >
    <h1
      className="title"
    >
      HATS
    </h1>
    <span
      className="subtitle"
    >
      SHOP NOW
    </span>
  </div>
</div>
`;
